### ISSUE TITLE
Missing MetricIDs after IndexDB rotation

### ISSUE THREAD

**Issue by fxrlv** [VictoriaMetrics/VictoriaMetrics#7417]
### Describe the bug

A few days after I had upgraded my vmstorage setup to v1.104.0-cluster I noticed that some of metrics (not all) were missed.
Since my vmstorage is configured with `-retentionPeriod=31d` that seems to be unexpected behavior for reading the last 4 days to return empty result.

After digging that a while I found that IndexDB had been rotated two days ago.
According to `vm_next_retention_seconds` IndexDB rotated at 30/10/2024 04:00 UTC — about a day after the version upgrade.

I tried to trace the query and here is what I saw:
- some storages reported finding 0 metric ids in cache and didn't go further to lookup prev indexdb
- the others reported `cache miss` and tried to lookup prev indexdb

That behavior seems to be legit and introduced by https://github.com/VictoriaMetrics/VictoriaMetrics/pull/7064

After restarting all vmstorages the issue have gone.

### To Reproduce

Not sure how to reproduce it since it seems related to indexdb rotation which is not easy to play around.

### Version

vmstorage-20241002-113739-tags-v1.104.0-cluster-0-gdfb2ad4ab4

### Logs

no errors, regular index and data merges 
```
2024-10-30 05:00:01.426	indexDB "/storage/indexdb/17E7EF2B84306F7B" has been dropped
2024-10-30 05:00:00.834	dropping indexDB "/storage/indexdb/17E7EF2B84306F7B"
```

### Screenshots

- current indexdb's cache miss
![image](https://github.com/user-attachments/assets/1c06cef2-7336-4d2d-8b41-231cc97d195c)

- current indexdb's cache hit
![image](https://github.com/user-attachments/assets/a4d38cb3-1b9f-4348-a781-e7ef3347598c)

### Used command-line flags

```
vmstorage
      -loggerFormat=json
      -loggerJSONFields=ts:timestamp,msg:message
      -storage.vminsertConnsShutdownDuration=0
      -storageDataPath=/storage
      -dedup.minScrapeInterval=15s
      -retentionPeriod=31d
      -enableTCP6
```

**Comment by f41gh7** [comment]
Could be related to the https://github.com/VictoriaMetrics/VictoriaMetrics/commit/787b9cd9a09f49ace10f23e1273eb7e21a3c463d

**Comment by f41gh7** [comment]
The bugfix was included into [v1.106.1](https://docs.victoriametrics.com/changelog/#v11061)
